lib/schema/utils.js

```diff
@@ -153,10 +156,370 @@ const parseNestedPropertyForResourceAndField = ({ arr, isValueArray }) => {
   }
 };
 
+/**
+ *
+ * @param {Object} obj
+ * @param {string} obj.string
+ * @param {string} obj.enums
+ * @param {string} obj.resourceName
+ * @param {string} obj.fieldName
+ * @param {Object} obj.schema
+ *
+ * @returns {string[]}
+ */
+const resolveStringList = ({ string, enums, schema, resourceName, fieldName }) => {
+  // if enum is of string type, no need to split
+  if (schema?.definitions?.MetadataMap?.[resourceName]?.[fieldName]?.lookupValues?.[string]?.isStringEnumeration) {
+    return [string];
+  }
+  if (enums.includes(string)) return [string];
+
+  // if type is anything other than Edm.String, we can split on commas
+  return string.split(',');
+};
+
+const addCustomValidationForEnum = ajv => {
+  ajv.removeKeyword('enum');
+
+  ajv.addKeyword({
+    keyword: 'enum',
+    validate: function validate(schema, data, _, ctx) {
+      const valid = [];
+      const nestedPayloadProperties = ctx?.instancePath?.split('/')?.slice(1) || [];
+
+      const resourceName = validationContext.getActiveResource();
+      const version = validationContext.getVersion();
+      const validationConfig = validationContext.getValidationConfig();
+      const activeSchema = validationContext.getSchema();
+      const isValueArray = validationContext.getPayloadType() === 'MULTI';
+
+      const { fieldName } = parseNestedPropertyForResourceAndField({
+        arr: nestedPayloadProperties,
+        isValueArray
+      });
+
+      if (typeof data === 'string') {
+        // Process the string to convert it into an array based on the enum definitions.
+        const parsedEnumValues = resolveStringList({
+          string: data,
+          enums: schema,
+          resourceName,
+          fieldName,
+          schema: activeSchema
+        });
+
+        // Validate each item against the schema's enum
+        for (const enumValue of parsedEnumValues) {
+          if (schema.includes(enumValue)) {
```

Comment:
On OData Simple Identifiers mentioned above, 

If the enum value is not a string type, then the OData Simple Identifier validation should be done.